src/canvas/SelectableCanvas.ts

```diff
@@ -667,44 +672,22 @@ export class SelectableCanvas<
    * @return {Boolean}
    */
   isTargetTransparent(target: FabricObject, x: number, y: number): boolean {
-    // in case the target is the activeObject, we cannot execute this optimization
-    // because we need to draw controls too.
-    if (isFabricObjectCached(target) && target !== this._activeObject) {
-      // optimizatio: we can reuse the cache
-      const normalizedPointer = this._normalizePointer(target, new Point(x, y)),
-        targetRelativeX = Math.max(
-          target.cacheTranslationX + normalizedPointer.x * target.zoomX,
-          0
-        ),
-        targetRelativeY = Math.max(
-          target.cacheTranslationY + normalizedPointer.y * target.zoomY,
-          0
-        );
-
-      return isTransparent(
-        target._cacheContext,
-        Math.round(targetRelativeX),
-        Math.round(targetRelativeY),
-        this.targetFindTolerance
-      );
-    }
-
-    const ctx = this.contextCache,
-      originalColor = target.selectionBackgroundColor,
-      v = this.viewportTransform;
-
-    target.selectionBackgroundColor = '';
-
+    const ctx = this.pixelFindContext;
     this.clearContext(ctx);
-
     ctx.save();
-    ctx.transform(v[0], v[1], v[2], v[3], v[4], v[5]);
+    ctx.translate(-x + this.targetFindTolerance, -y + this.targetFindTolerance);
+    ctx.transform(...this.viewportTransform);
+    const selectionBgc = target.selectionBackgroundColor;
+    target.selectionBackgroundColor = '';
     target.render(ctx);
+    target.selectionBackgroundColor = selectionBgc;
     ctx.restore();
-
-    target.selectionBackgroundColor = originalColor;
-
-    return isTransparent(ctx, x, y, this.targetFindTolerance);
+    return isTransparent(
+      ctx,
+      this.targetFindTolerance,
+      this.targetFindTolerance,
+      this.targetFindTolerance
```

Comment:
looking at this again shouldn't it be multiplied by retina?

Reply:
in my branch i completely removed retina from the logic of target transparent.
I don't think it should be involved at all.

Reply:
I thought it is needed for precision in case of blur

Reply:
considering retina makes the cached/noncached case very similar and cut differences there.
But force us to continuosly round numbers

Reply:
That isn't too bad, is it?
I am thinking that this feature is used with text and text is heavily impacted by retina

Reply:
yes i think it will win the smaller code at the end.
traget transparency isn't meant to be a pixel sniper effect. It has just to guess if is completely transparent or not.